volmdlr/wires.py

```diff
@@ -168,26 +168,65 @@ def point_belongs(self, point, abs_tol=1e-7):
         for i, primitive in enumerate(self.primitives):
             belongs = primitive.point_belongs(point, abs_tol=abs_tol)
             if belongs:
-                return i
+                return True
         return False
 
     def abscissa(self, point):
         '''
         compute the curvilinear abscisse of a point on a wire
         '''
 
-        index = self.point_belongs(point)
-
-        length = self.primitives[index].abscissa(point)
-        for primitive in self.primitives[0:index]:
-            length += primitive.length()
+        if self.point_belongs(point, 1e-6):
+            length = 0
+            for primitive in self.primitives:
+                if primitive.point_belongs(point, 1e-6):
+                    length += primitive.abscissa(point)
+                    break
+                length += primitive.length()
+            return length
 
-        return length
+        raise ValueError('Point is not on wire')
 
     def sort_points_along_wire(self, points):
 
         return sorted(points, key=lambda point: self.abscissa(point))
 
+    @classmethod
+    def from_wires(cls, wires):
+        '''
+        define a wire from successive wires
+        '''
+
+        primitives = []
+        for wire in wires:
+            primitives.extend(wire.primitives)
+
+        wire = cls(primitives)
+
+        if not wire.is_ordered():
+            return wire.order_wire()
+        return wire
+
+    def inverted_primitives(self):
```

Comment:
why not call reverse?

Reply:
I'd used invert instead of reverse because, I'd found that the word 'invert' is used for Contour. So, I tried to use the same terminology!